index.js

```diff
@@ -118,93 +149,146 @@ async function isNotLatestTerraformVersion() {
+  }
+  return logs;
+}
+
+async function getContents(owner, repo, path) {
   const { data: data } = await octokit.repos.getContents({
+    owner: owner,
+    repo: repo,
+    path: path,
+  });
+  return data;
+}
+
+async function createPR(
+  projects,
+  issueContent,
+  releaseConfig,
+  newReleaseContentBlob
+) {
+  const branchName = `release-${new Date().getTime()}`;
+  const manifestsSha = await getHeadSha("notification-manifests", "main");
+  const logs = await buildLogs(projects);
+
+  const ref = await octokit.git.createRef({
     owner: "cds-snc",
     repo: "notification-manifests",
-    path: "env/production/kustomization.yaml",
+    ref: `refs/heads/${branchName}`,
+    sha: manifestsSha,
   });
 
-  const { data: issueData } = await octokit.repos.getContents({
+  const manifestUpdates = projects
+    .map((project) => {
+      return `${project.name}:${shortSha(project.headSha)}`;
+    })
+    .join(" and ");
+  const update = await octokit.repos.createOrUpdateFile({
     owner: "cds-snc",
     repo: "notification-manifests",
-    path: ".github/PULL_REQUEST_TEMPLATE.md",
+    branch: branchName,
+    sha: releaseConfig.sha,
+    path: "env/production/kustomization.yaml",
+    message: `Updated manifests to ${manifestUpdates}`,
+    content: newReleaseContentBlob,
   });
 
-  const apiSha = await getHeadSha("notification-api");
-  const adminSha = await getHeadSha("notification-admin");
-  const manifestsSha = await getHeadSha("notification-manifests", "main");
-
-  const fileContent = YAML.parse(Base64.decode(data.content));
-  const issueContent = Base64.decode(issueData.content);
-
-  fileContent.images.forEach((image) => {
-    if (image.name == "admin") {
-      oldAdminSha = image.newName.split(":").slice(-1)[0]; // eslint-disable-line no-unused-vars, no-undef
-      image.newName = `${AWS_ECR_URL}/notify-admin:${adminSha.slice(0, 7)}`;
-    }
-    if (image.name == "api") {
-      oldApiSha = image.newName.split(":").slice(-1)[0]; // eslint-disable-line no-unused-vars, no-undef
-      image.newName = `${AWS_ECR_URL}/notify-api:${apiSha.slice(0, 7)}`;
-    }
+  const pr = await octokit.pulls.create({
+    owner: "cds-snc",
+    repo: "notification-manifests",
+    title: `[AUTO-PR] Automatically generated new release ${new Date().toISOString()}`,
+    head: branchName,
+    base: "main",
+    body: issueContent.replace(
+      "> Give details ex. Security patching, content update, more API pods etc",
+      logs
+    ),
+    draft: true,
   });
+  return Promise.all([ref, update, pr]);
+}
 
-  const newBlob = Base64.encode(YAML.stringify(fileContent));
+async function hydrateWithSHAs(releaseConfig, projects) {
+  return await Promise.all(
+    releaseConfig.images.map(async (image) => {
+      const matchingProject = projects.find((project) =>
+        image.newName.includes(project.ecrName)
```

Comment:
`startsWith` instead of `includes`?

Reply:
A `newName` looks like `public.ecr.aws/cds-snc/notify-admin:c779ddf` so we need to look somewhere in the middle. It would be more ideal to use a regex to extract the proper name at the exact location... Typescript would be nice to use the extra class/typing features it offers. 

Another nice improvement would be to have matching names between GitHub and Elastic Cloud Repositories so we wouldn't need to map different names between the two repository types.